projects/rero/ng-core/src/lib/record/detail/detail.component.ts

```diff
@@ -73,22 +123,29 @@ export class DetailComponent implements OnInit {
   }
 
   /**
-   * Load record from API
+   * Delete the record and go back to previous page.
+   * @param event - DOM event
+   * @param pid - string, PID to remove
    */
-  private getRecord() {
-    const pid = this.route.snapshot.paramMap.get('pid');
-    const type = this.route.snapshot.paramMap.get('type');
+  public deleteRecord(event: Event, pid: string) {
+    event.preventDefault();
 
-    this.recordService.getRecord(type, pid)
-      .subscribe(
-        (record) => {
-          this.record = record;
-          this.loadRecordView();
-        },
-        (error) => {
-          this.error = error;
-        }
-      );
+    this.recordUiService.deleteRecord(pid, this.route.snapshot.paramMap.get('type')).subscribe((result: any) => {
+      if (result === true) {
+        this.location.back();
+      }
+    });
+  }
+
+  /**
+   * Show a modal containing message given in parameter.
+   * @param event - DOM event
+   * @param message - message to display into modal
+   */
+  public showDeleteMessage(event: Event, message: string) {
+    event.preventDefault();
```

Comment:
is it really required?

Reply:
It's required when a link is clicked. I converted links to buttons.